### ISSUE TITLE
[Question] Adding separate optimizers/loss functions per network

### ISSUE THREAD

**Issue by jren03** [danijar/dreamerv3#121]
Hello,

If I would like to experiment with different optimizers and loss functions on each of the different modules, how would I go about doing this?

From what I've gathered so far, I would need to remove those modules from the list [here](https://github.com/danijar/dreamerv3/blob/e97b6329b55392a5a5313160d24ad1202b13f2c5/dreamerv3/agent.py#L100) and also declare a new optimizer accordingly. Then, I would need to implement a separate `train()` and `loss()` function in the style you have in agent.py. 

I am less sure on how modifications should be made in jaxagent.py. Specifically, if I am training the separated modules at different intervals, would I need to create a new set of `self.params`, `self.policy_keys`, and `self.policy_params` for each module that I separate out? ([Reference](https://github.com/danijar/dreamerv3/blob/e97b6329b55392a5a5313160d24ad1202b13f2c5/dreamerv3/jaxagent.py#L71C1-L83C30)) 

Since I have a new `new_train()` function in agent.py, I would need to have a matching `new_train()`in jaxagent.py, and thus need to address the lines [here](https://github.com/danijar/dreamerv3/blob/e97b6329b55392a5a5313160d24ad1202b13f2c5/dreamerv3/jaxagent.py#L174C1-L179C43):

```python
  @embodied.timer.section('jaxagent_train')
  def train(self, data, carry):
    seed = data['seed']
    data = self._filter_data(data)
    allo = {k: v for k, v in self.params.items() if k in self.policy_keys}
    dona = {k: v for k, v in self.params.items() if k not in self.policy_keys}
    with embodied.timer.section('jit_train'):
      with self.train_lock:
        self.params, outs, carry, mets = self._train(
            allo, dona, data, carry, seed)
```

Thank you for your help!

**Comment by danijar** [comment]
You could also just use a single optimizer class and apply different updates to each group of parameters. There is support for different learning rate groups in the code already, and it would look similar to that. And of course you can stop gradients where you like.
